server/src/main/java/org/opensearch/cluster/service/ClusterManagerTaskThrottler.java

```diff
@@ -128,10 +133,21 @@ void validateSetting(final Settings settings) {
         }
     }
 
-    void updateSetting(final Settings settings) {
-        Map<String, Settings> groups = settings.getAsGroups();
-        for (String key : groups.keySet()) {
-            updateLimit(key, groups.get(key).getAsInt("value", MIN_THRESHOLD_VALUE));
+    void updateSetting(final Settings newSettings) {
+        Map<String, Settings> groups = newSettings.getAsGroups();
+        Set<String> settingKeys = new HashSet<>();
+        // Adding keys which are present in new Setting
+        settingKeys.addAll(groups.keySet());
+        // Adding already configured keys as well,
+        // we might need to set it to default value if it's value is null in new setting
+        settingKeys.addAll(tasksThreshold.keySet());
+        for (String key : settingKeys) {
+            Settings setting = groups.get(key);
+            if (setting == null) {
```

Comment:
Maybe we can use `.orElse` kind of thing.

Reply:
Here if setting is not null we want to get specific value from it, so `orElse` might not work here.

To improve readability, I have used ternary operator, let me know if that helps.